src/OVAL/probes/independent/system_info.c

```diff
@@ -535,7 +681,35 @@ static char * _offline_chroot_get_hname(void)
 }
 #endif
 
-void *probe_init(void)
+#ifdef _WIN32
+static char *get_windows_version()
+{
+	/* We can't use GetVersionEx to get Windows version because we want to get
+	 * the real Windows version, but the compatibilty mode lies to applications
+	 * about system version. Moreover the GetVersionEx function is deprecated
+	 * since Windows 8.1. Instead we need to access RtlGetVersion.
+	 * Code inspired by https://github.com/GNOME/glib/blob/master/glib/gwin32.c
+	 */
+	OSVERSIONINFOEXW osverinfo;
+	typedef NTSTATUS(WINAPI fRtlGetVersion) (PRTL_OSVERSIONINFOEXW);
+	fRtlGetVersion *RtlGetVersion;
+	HMODULE hmodule;
+
+	hmodule = LoadLibraryW(L"ntdll.dll");
+	RtlGetVersion = (fRtlGetVersion *)GetProcAddress(hmodule, "RtlGetVersion");
+
+	memset(&osverinfo, 0, sizeof(OSVERSIONINFOEXW));
+	osverinfo.dwOSVersionInfoSize = sizeof(OSVERSIONINFOEXW);
+	RtlGetVersion(&osverinfo);
+
+	char *version = malloc(VERSION_LEN);
+	snprintf(version, VERSION_LEN, "%d.%d.%d", osverinfo.dwMajorVersion, osverinfo.dwMinorVersion, osverinfo.dwBuildNumber);
```

Comment:
any reason not to use oscap_sprintf?

Reply:
I always forget about existence of this function :smile: